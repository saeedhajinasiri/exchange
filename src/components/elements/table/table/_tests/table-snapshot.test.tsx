import React from "react";
import {cleanup, render} from "test";
import Table from '../table'
import {TableHead} from "@elements";

afterEach(cleanup)

describe('Test Table UI', () => {
  it('Default', () => {
    const {asFragment} = render(<Table/>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Test with className prop', () => {
    const {asFragment} = render(<Table className={'test-class-name'}/>);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Test Table Colors', () => {
  const children = <TableHead/>
  it('color=primary', () => {
    const {asFragment} = render(<Table color={"primary"}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=secondary', () => {
    const {asFragment} = render(<Table color={"secondary"}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=tertiary', () => {
    const {asFragment} = render(<Table color={'tertiary'}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=grey', () => {
    const {asFragment} = render(<Table color={'grey'}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=light', () => {
    const {asFragment} = render(<Table color={'light'}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('color=dark', () => {
    const {asFragment} = render(<Table color={'dark'}>{children}</Table>);
    expect(asFragment()).toMatchSnapshot();
  });
});
