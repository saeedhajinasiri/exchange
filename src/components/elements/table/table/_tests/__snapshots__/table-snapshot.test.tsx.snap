// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Table Colors color=dark 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="dark"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table Colors color=grey 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="grey"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table Colors color=light 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="light"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table Colors color=primary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="primary"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table Colors color=secondary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="secondary"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table Colors color=tertiary 1`] = `
<DocumentFragment>
  <table
    class="w-full"
    color="tertiary"
  >
    <thead
      class="table-header-group"
    />
  </table>
</DocumentFragment>
`;

exports[`Test Table UI Default 1`] = `
<DocumentFragment>
  <table
    class="w-full"
  />
</DocumentFragment>
`;

exports[`Test Table UI Test with className prop 1`] = `
<DocumentFragment>
  <table
    class="w-full test-class-name"
  />
</DocumentFragment>
`;
